lib/src/main/java/io/ably/lib/realtime/Presence.java

```diff
@@ -26,37 +26,49 @@
 	 ************************************/
 
 	/**
-	 * Get the presence state for this Channel. Implicitly attaches the
-	 * Channel. However, if the channel is in or moves to the FAILED
-	 * state before the operation succeeds, it will result in an error
-	 * @return: the current present members.
+	 * String parameter names to get() call with Param[] as an argument
+	 */
+	public final static String GET_WAITFORSYNC = "waitForSync";
+	public final static String GET_CLIENTID = "clientId";
+	public final static String GET_CONNECTIONID = "connectionId";
+
+	/**
+	 * Get the presence state for this channel. Take Param[] array as an argument
+	 * @param params
+	 * @return
 	 * @throws AblyException
+	 * @throws InterruptedException
 	 */
-	public synchronized PresenceMessage[] get() throws AblyException {
+	public synchronized PresenceMessage[] get(Param... params) throws InterruptedException, AblyException {
```

Comment:
I concur with @paddybyers here, I would prefer to avoid this exceptional exception being thrown here when everywhere else (and certainly in either client libs) we try and stick with Ably exceptions.